src/Hunk/SplitHunk/SplitChange.js

```diff
@@ -78,140 +78,132 @@ const renderCells = args => {
     ];
 };
 
-export default class SplitChange extends PureComponent {
-
-    static propTypes = {
-        oldSelected: PropTypes.bool.isRequired,
-        newSelected: PropTypes.bool.isRequired,
-        oldTokens: PropTypes.arrayOf(PropTypes.object),
-        newTokens: PropTypes.arrayOf(PropTypes.object),
-    };
-
-    static defaultProps = {
-        oldTokens: null,
-        newTokens: null,
-    };
-
-    state = {
-        hover: '',
-    };
-
-    constructor(props) {
-        super(props);
-
-        const unmarkHover = () => this.setState({hover: ''});
-        const markHoverAs = side => () => this.setState({hover: side});
-        const ownEventsOnOldSide = {
-            onMouseEnter: markHoverAs('old'),
+const SplitChange = props => {
+    const {
+        className,
+        gutterClassName,
+        codeClassName,
+        gutterEvents,
+        codeEvents,
+        oldChange,
+        newChange,
+        oldSelected,
+        newSelected,
+        oldTokens,
+        newTokens,
+        monotonous,
+        hideGutter,
+        generateAnchorID,
+        gutterAnchor,
+        renderToken,
+    } = props;
+
+    const [hover, setHover] = useState('');
+    const unmarkHover = useCallback(() => setHover(''), []);
+    const useCallbackOnSide = (side, customCallbacks) => {
```

Comment:
Define a hook inside a component?

Reply:
This is what I also consider weird, though it works since it is just a simple function combining some other hook calls, I tried to move it outside but it should get a very large amount of parameters

Reply:
Yes, I'm sure it works.

I think the problem is there is not any hook to get the props of the Component.

Also it is why `react-redux` can't simply `useConnect(oldMapStateToProps)`. It must be `useConnect(props, oldMapStateToProps)`